messaging.rst

```diff
@@ -159,202 +159,107 @@ Locked Transfer message
-
-After creating the packed form of the data we can use ``keccak256`` to create the ``additional_hash``. ::
-
-    additional_hash = keccak256(packed_message_data)
-
-    0xaeba3609fa01beca6b0b54f49780f301d59a8994bc0b383b2571eee0e6dacad4
-
-3. Balance Hash
+The data will be packed as follows to compute the :term:`additional hash`:
+
++--------------------------------------+---------+-------------+
+| Field                                | Type    | Size (bytes)|
++======================================+=========+=============+
+| command_id (7 for ``LockedTransfer``)| uint8   |   1         |
++--------------------------------------+---------+-------------+
+| message_identifier                   | uint64  |   8         |
++--------------------------------------+---------+-------------+
+| payment_identifier                   | uint64  |   8         |
++--------------------------------------+---------+-------------+
+| expiration                           | uint256 |  32         |
++--------------------------------------+---------+-------------+
+| token_network_address                | address |  20         |
++--------------------------------------+---------+-------------+
+| token                                | address |  20         |
++--------------------------------------+---------+-------------+
+| recipient                            | address |  20         |
++--------------------------------------+---------+-------------+
+| target                               | address |  20         |
++--------------------------------------+---------+-------------+
+| initiator                            | address |  20         |
++--------------------------------------+---------+-------------+
+| secrethash                           | bytes32 |  32         |
++--------------------------------------+---------+-------------+
+| amount                               | uint256 |  32         |
++--------------------------------------+---------+-------------+
+| fee                                  | uint256 |  32         |
++--------------------------------------+---------+-------------+
+| metadata_hash                        | bytes32 |  32         |
++--------------------------------------+---------+-------------+
+
+The computation of the ``metadata_hash`` as well as the exact format of the ``metadata`` itself
+are implementation specific.
+
+This will be used to generate the the data field called ``additional_hash``, which is a required
+part of the process to create the message signature. It is computed as the ``keccak256``-hash
```

Comment:
I'm not sure where to put it, but it would be nice to have an explanation somewhere why we have the `additional_hash` in the first place. The reason is that it allows as to notice if any field has been tampered with (because the `additional_hash` would be different and thus the overall signature) while still having a signature that will be accepted by the contract (because it uses the additional hash only for the signature check and nothing else).

Other approaches would require two signatures, which would make this process less efficient.

Reply:
I thought that explanation was already in `terminology.rst`, will check.